### ISSUE TITLE
PUT App API fails with 400 because consent_method is not supplied

### ISSUE THREAD

**Issue by mtttcgcg** [okta/terraform-provider-okta#188]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.13.0
provider registry.terraform.io/terraform-providers/okta v3.4.1
AND
Terraform v0.13.3
provider registry.terraform.io/oktadeveloper/okta v3.5.0

### Affected Resource(s)

* okta_app_oauth

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    okta = {
      source = "oktadeveloper/okta"
    }
  }
}
provider "okta" {
   org_name  = "your_org"
   base_url  = "oktapreview.com"
   api_token = "SSWSToken"
}

resource okta_app_oauth example {
  label          = "MyTerraformApp"
  type           = "web"
  grant_types    = ["authorization_code"]
  redirect_uris  = ["https://example.com/"]
  response_types = ["code"]
  lifecycle {
    ignore_changes = [groups]
  }
}
```

### Debug Output

https://gist.github.com/mtttcgcg/013ed47b86a79ff1f8d1d5ba3c408925

### Panic Output

None - no crash

### Expected Behavior

Editing an app should succeed.

### Actual Behavior

Updating apps fails because the existing state of the app is not retrieved with a GET before constructing the  PUT request. This results is missing fields in the PUT request, such as consent_method.

### Steps to Reproduce

1) Using the above main.tf file, run `terraform apply --auto-approve`
2) Edit any field of the app, such as adding a 2nd redirect URI, such that it now contains : ["https://example.com/ ,"http://url2"]
3) run `terraform apply --auto-approve`

### Important Factoids

I contacted Okta support about this, and they said the plugin should make a GET request to get current state, then substitute the changed items, before making its PUT request. 

### References

* https://developer.okta.com/docs/reference/api/apps/

**Comment by jeremie0** [comment]
similar issue https://github.com/oktadeveloper/terraform-provider-okta/issues/122

**Comment by bogdanprodan-okta** [comment]
Hi @mtttcgcg! Thanks for creating this issue. Could you please verify if this issue still occurs?

**Comment by bogdanprodan-okta** [comment]
Hi @mtttcgcg! The issue should be fixed in the latest 3.7.2 version. In case it still exists, please reopen this issue.
